### ISSUE TITLE
raw flow-graph data

### ISSUE THREAD

**Issue by alston2506** [sbustreamspot/sbustreamspot-data#1]
Could you please provide the raw flow-graph data so that I can carry out my own research?

Thx!

**Comment by emaadmanzoor** [comment]
Sure! Please find the raw data here (graciously hosted by Caleb Belth at the University of Michigan): https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing

Feel free to re-open this issue if you have further questions, I am happy to help.

**Comment by alston2506** [comment]
How can I get the more precise timestamp of each event entry? such as in what time of which day the event happened？
By the way, are all the events that happened in one host? If not, could you please provide more host information of your laboratory environment？

**Comment by emaadmanzoor** [comment]
Below is a note from my co-author Sadegh M. Milajerdi about the format of this CSV file (he generated the data for us):

```
Each tuple of the dataset is in the following format:
Source id , source type, Destination id, destination type, Edge type, Timestamp, graphid(which is like a tag)

Source types are: Process, Thread
Destination Types are: File, Socket, Process, Thread
And I tried to cover all Edge types that we are interested based on Sekar’s data model consisting of:
fork, clone, waitpid, open, read, write, pwrite, writev, pwritev, close, bind, connect, listen, accept, send, sendto, sendmsg, recv, recvfrom, recvmsg, fstat, ftruncate, mmap2, access, link, unlink, stat, chmod, chown, truncate, execl, execv, execle, execve, execlp, execvp.

As you see, I have only covered about 40 events that seems to be more important than the other system calls.
Graphids differentiate use cases from each other.
Time is shown in nanoseconds from the moment the firefox starts. So it is relational and all use_cases interleave each other in time.
The timestamp shown for each tuple, is the moment that the event is passed to kernel not the moment that the kernel returns.
In some tuples, different graph ids may use same object ids which means two different use cases are accessing same object. 
```
The data was generated using SystemTap on a server to collect system call traces, and running some tasks in Firefox (which we call "scenarios"). For example, the "YouTube" scenario involved starting the machine, starting SystemTap, opening Firefox and browsing YouTube and then stopping Firefox. The scenarios that each graph ID corresponds to are described in the sbustreamspot-data README.md.

Sadegh would be the best person to email about this, but please chime in with any further questions! I will do my best to help out.

**Comment by alston2506** [comment]
What still troubles me is that in graph #0, for example, the timestamp of the last tuple is just 31908302156ns(≈31.9s)，but there are 104968 events in graph #0, it‘s too much in my opinion.  Does the server records other system call traces irrelevant to the task scenario？ or there are multiple users perform the task scenario in the single one server simultaneously？

**Comment by emaadmanzoor** [comment]
You are right that all system call traces running on the machine are recorded (including ones not from Firefox). This was done to construct a realistic dataset for anomaly detection, since attacks may not take place in isolation.

However, there was only one user on the system at that time.

**Comment by alston2506** [comment]
Okay, I understand！
Appreciate your answer！

**Comment by lizitong67** [comment]
One more question.
What‘s the meaning of the data entry whose id is -1 and type is MAP_ANONYMOUS?

**Comment by emaadmanzoor** [comment]
That's a type of memory mapping in UNIX-based systems, please take a look
at: https://man7.org/linux/man-pages/man2/mmap.2.html

**Comment by Irish-kw** [comment]
HI Emaad :

Sorry for asking a stupid question.
But I wonder to know why the source node is always the same for each scenario's subset.
For example, 

![image](https://github.com/sbustreamspot/sbustreamspot-data/assets/49675917/850251e1-52ed-4d9f-8889-2c009977ac85)

the source node always is 4, when I plot a graph, it would be like a "Virus" structure.
Am I missing something?  Thank you for your help!

**Comment by emaadmanzoor** [comment]
The source is typically the main process (eg. a web browser) that forks
additional processes (eg. extensions) to perform some tasks. Most system
calls will originate from the main process, which is why most edges you
will see will be from a single source. This is just how a lot of processes
look like when you construct a graph of their system calls.

**Comment by Irish-kw** [comment]
HI Emaad :

Thank you for your response to my stupid question.

BR.

**Comment by Irish-kw** [comment]
HI Emaad :

Sorry for asking the question again, but how could I know which node is attacked in the attack dataset ?
Thank you for your help.

BR.

**Comment by emaadmanzoor** [comment]
There is no concept of attacked node, the entire graph is an "attack graph" if the graph was generated as a result of an attack scenario. The nodes are just pieces of software (eg. a browser, a "wget" process, etc.).

**Comment by Irish-kw** [comment]
HI Emaad :

Thank you for answering my stupid question again!

BR.

**Comment by dlgdwy** [comment]
Excuse me, would you like to ask whether the graph id identifies these data on a graph? Does the compressed package you provided for the ALL dataset contain graphic data with id 0 to 199?

**Comment by emaadmanzoor** [comment]
Yes the graph ID identifies the graph. The ALL compressed package contains
graphs with IDs 0 to199.

**Comment by 026plan** [comment]
> Sure! Please find the raw data here (graciously hosted by Caleb Belth at the University of Michigan): https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing确定！请在此处找到原始数据（由密歇根大学的 Caleb Belth 慷慨主持）：https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing
> 
> Feel free to re-open this issue if you have further questions, I am happy to help.如果您还有其他问题，请随时重新打开此问题，我很乐意为您提供帮助。

Hello, could you please re-provide the original system log data? The provided network disk link has expired.Thank you.

**Comment by DanyuSun** [comment]
> > Sure! Please find the raw data here (graciously hosted by Caleb Belth at the University of Michigan): [https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing确定！请在此处找到原始数据（由密歇根大学的](https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing%E7%A1%AE%E5%AE%9A%EF%BC%81%E8%AF%B7%E5%9C%A8%E6%AD%A4%E5%A4%84%E6%89%BE%E5%88%B0%E5%8E%9F%E5%A7%8B%E6%95%B0%E6%8D%AE%EF%BC%88%E7%94%B1%E5%AF%86%E6%AD%87%E6%A0%B9%E5%A4%A7%E5%AD%A6%E7%9A%84) Caleb Belth 慷慨主持）：https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing
> > Feel free to re-open this issue if you have further questions, I am happy to help.如果您还有其他问题，请随时重新打开此问题，我很乐意为您提供帮助。
> 
> Hello, could you please re-provide the original system log data? The provided network disk link has expired.Thank you.

Yes, same problem. Could you please update the link? Thank you.

**Comment by Namtayto** [comment]
> > > Sure! Please find the raw data here (graciously hosted by Caleb Belth at the University of Michigan): [https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing确定！请在此处找到原始数据（由密歇根大学的](https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing%E7%A1%AE%E5%AE%9A%EF%BC%81%E8%AF%B7%E5%9C%A8%E6%AD%A4%E5%A4%84%E6%89%BE%E5%88%B0%E5%8E%9F%E5%A7%8B%E6%95%B0%E6%8D%AE%EF%BC%88%E7%94%B1%E5%AF%86%E6%AD%87%E6%A0%B9%E5%A4%A7%E5%AD%A6%E7%9A%84) Caleb Belth 慷慨主持）：https://drive.google.com/file/d/1X-1vpo59_kuIBWi8-baRPOb1o2umW5rH/view?usp=sharing
> > > Feel free to re-open this issue if you have further questions, I am happy to help.如果您还有其他问题，请随时重新打开此问题，我很乐意为您提供帮助。
> > 
> > 
> > Hello, could you please re-provide the original system log data? The provided network disk link has expired.Thank you.
> 
> Yes, same problem. Could you please update the link? Thank you.

I found this:

https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/83KYJY

**Comment by Win7ery** [comment]
> Yes, same problem. Could you please update the link? Thank you.

Yes, same problem. Could you please update the link? Thank you.

**Comment by kiankyars** [comment]
let me know too please if the link has been updated
